spec/datadog/profiling/collectors/discrete_dynamic_sampler_spec.rb

```diff
@@ -6,7 +6,7 @@
 
   before do
     skip_if_profiling_not_supported(self)
-    @now = Time.now.to_f
+    @now = Time.now.to_r
```

Comment:
I'm not sure this change is needed -- in fact, this spec works as well with `to_i`, and looking at the code it kinda makes sense since `@now` is used to calculate time differences so any point in time is expected to yield the same results.

Reply:
Uh yeah remnant from some troubleshooting I was doing on an earlier flaky version of a test (floats will approximate the timings from `Time.now` so I tried to keep to rationals to ensure no approximation was done. but then I simply rewrote the test to not care about it 😂 )